core/domain/html_validation_service.py

```diff
@@ -887,23 +887,28 @@ def get_invalid_svg_tags_and_attrs(svg_string):
         svg_string: str. The SVG string.
 
     Returns:
-        tuple. A 2-tuple, the first element of which is a list of invalid tags,
-        and the second element of which is a list of invalid attributes.
+        tuple(list(str), list(str)). A 2-tuple, the first element of which
+        is a list of invalid tags, and the second element of which is a
+        list of invalid tag-specific attributes.
+        eg. (['invalid-tag1', 'invalid-tag2'], ['path:invalid-attr'])
     """
     soup = bs4.BeautifulSoup(svg_string.encode('utf-8'), 'html.parser')
     invalid_elements = []
     invalid_attrs = []
     for element in soup.find_all():
-        if element.name.lower() in constants.ALLOWED_SVG_TAGS:
+        if element.name.lower() in constants.SVG_ATTRS_WHITELIST.keys():
             for attr in element.attrs:
-                if attr.lower() not in constants.ALLOWED_SVG_ATTRS:
-                    invalid_attrs.append(attr)
+                if attr.lower() not in (
+                        constants.SVG_ATTRS_WHITELIST[element.name.lower()]):
+                    invalid_attrs.append('%s:%s' % (element.name, attr))
         else:
             invalid_elements.append(element.name)
+            for attr in element.attrs:
```

Comment:
Don't know that you need this part if the entire element is invalid.

Reply:
I'm fine with removing this part. My initial thought was that, as part of validation, all errors need be reported.

Reply:
Ah, but if the error is that the tag name is wrong, the attr names don't matter.